src/QUICConnection.ts

```diff
@@ -471,12 +461,24 @@ class QUICConnection extends EventTarget {
 
   /**
    * This will set up the connection initiate sending
+   * @param initialData - If the connection is server initiated then the data that initiated it needs to be provided here
    */
-  public async start(): Promise<void> {
+  public async start(initialData?: {
+    data: Uint8Array;
+    remoteInfo: RemoteInfo;
+  }): Promise<void> {
     this.logger.info(`Start ${this.constructor.name}`);
     // Set the connection up
     this.socket.connectionMap.set(this.connectionId, this);
-    await this.send();
+    await withF(
+      [contextsUtils.monitor(this.lockbox, RWLockWriter)],
```

Comment:
You're already using `withF` here. So why not continue using it instead of the custom utility? Or vice versa?

Reply:
Don't use `initialData` you're basing this on a server type originally. I think it should be made clear that it is done only on server.

Reply:
`withF` is used here because the monitor is always created here. The utility is used when the `mon` is optional and may need to be created and used. I would've inlined it but branching the monitor creation is a little annoying outside of a separate function.

Made clear as in use a comment? Make the `initialData` a property of the QUICConnection and check that? Throw an exception if the connection type isn't `server` and data is provided?